pkg/stacks/truncate/truncate_test.go

```diff
@@ -0,0 +1,200 @@
+// predictable way offering safe values usable in deterministic field searches.
+package truncate
+
+import (
+	"strings"
+	"testing"
+)
+
+var (
+	str63  = strings.Repeat("1234567890", 6) + "123"
+	str253 = strings.Repeat("1234567890", 25) + "123"
+)
+
+func Test_truncate(t *testing.T) {
+	type args struct {
+		str          string
+		length       int
+		suffixLength int
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    string
+		wantErr bool
+	}{
+		{
+			name:    "StrTooShort",
+			args:    args{str: "a", length: 0, suffixLength: 0},
+			want:    "",
+			wantErr: true,
+		},
+		{
+			name:    "SuffixExceedsStr",
+			args:    args{str: "a", length: 0, suffixLength: 2},
+			want:    "",
+			wantErr: true,
+		},
+		{
+			name:    "SuffixExceedsSum",
+			args:    args{str: strings.Repeat("12345678890", 21), length: 200, suffixLength: 100},
+			want:    "",
+			wantErr: true,
+		},
+		{
+			name:    "SuffixTooShort",
+			args:    args{str: "12345678901", length: 10, suffixLength: 1},
+			want:    "",
+			wantErr: true,
+		},
+		{
+			name:    "StrNotTruncated",
+			args:    args{str: "1234567890", length: 10, suffixLength: 1},
+			want:    "1234567890",
+			wantErr: false,
+		},
+		{
+			name:    "StrTruncated",
+			args:    args{str: "12345678901", length: 10, suffixLength: 5},
+			want:    "12345-ezw4",
+			wantErr: false,
+		},
+		{
+			name:    "AllTruncated",
+			args:    args{str: "1234567890", length: 5, suffixLength: 5},
+			want:    "-agzq",
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := Truncate(tt.args.str, tt.args.length, tt.args.suffixLength)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("truncate() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("truncate() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestLabelName(t *testing.T) {
+	type args struct {
+		str string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "short",
+			args: args{str: "foo"},
+			want: "foo",
+		},
+		{
+			name: "max",
+			args: args{str: str63},
+			want: str63,
+		},
+		{
+			name: "truncate",
+			args: args{str: str253},
+			want: strings.Repeat("1234567890", 5) + "1234567-ij43w",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := LabelName(tt.args.str); got != tt.want {
+				t.Errorf("LabelName() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestLabelValue(t *testing.T) {
+	type args struct {
+		str string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "short",
+			args: args{str: "foo"},
+			want: "foo",
+		},
+		{
+			name: "max",
+			args: args{str: str63},
+			want: str63,
+		},
+		{
+			name: "truncate",
+			args: args{str: str253},
+			want: strings.Repeat("1234567890", 5) + "1234567-ij43w",
+		},
```

Comment:
nit (not blocking): it could be nice to add another sanity (and easy to read) test case for one of the real life scenarios we saw that we are hoping to fix with this new truncation behavior.  These cases test the major scenarios we care about, but they may be a bit hard to read since the last case here has just a little bit of "magic" in them.

Reply:
I'll add a test case we can refer to as an example.

The testing example in the PR description is similarly obtuse.  The digits make it easy to visually count the character length but don't reasonably convey how this affects "my-really-really-long-namespace.stack-name-is-also-long-version-one-point-oh"